### ISSUE TITLE
Tabs [a11y]: Interactive controls must not be nested (Violation)

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#13886]
### Package

@carbon/react

### Browser

Chrome, Safari, Firefox

### Operating System

MacOS

### Package version

1.30.0

### Automated testing tool and ruleset

Storybook Accessibility tab

### Description

The error was reported on [slack](https://gbsla.slack.com/archives/C2K6RFJ1G/p1685367757498979).

There is a violation that says "Interactive controls must not be nested"

![Screenshot 2023-05-29 at 15 27 07](https://github.com/carbon-design-system/carbon/assets/52183462/09d42754-ba81-4e8c-a8a5-f9d76e0fecc9)

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tabs--manual

### Steps to reproduce

Go to this [storybook link](https://react.carbondesignsystem.com/?path=/story/components-tabs--manual) and click on the Accessibility tab

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
@guidari can you reproduce this with accessibility-checker and include what dom element this is referring to?

**Comment by guidari** [comment]
> https://react.carbondesignsystem.com/?path=/story/components-tabs--manual

When testing with the accessbility-checker it doesn't show the violation. Maybe it is something only in storybook...

Should I close that @tay1orjones ?

**Comment by btepe** [comment]
@tay1orjones We encountered the same a11y issue in our apps using the axe a11y check, which is very commonly used for automated testing. The fact that a div with `role='button'` is nested inside the actual tab `button` element violates the following a11y rule:
<img width="1280" alt="Screenshot 2023-08-31 at 14 49 56" src="https://github.com/carbon-design-system/carbon/assets/24353185/cdefcf6b-1e75-4d1f-b221-24eec0fd77d9">

Here you can see the DOM elements that it is referring to: 
<img width="532" alt="Screenshot 2023-08-31 at 14 50 18" src="https://github.com/carbon-design-system/carbon/assets/24353185/b45340a5-111d-4c78-a973-7291e53aa957">

And this is the DOM tree view with the nested element being highlighted, for some more context:
<img width="351" alt="Screenshot 2023-08-31 at 14 57 26" src="https://github.com/carbon-design-system/carbon/assets/24353185/44cce13b-a890-49ec-8a43-b7a6f1ce1e2a">

It can be reproduced by going to the [Carbon react storybook Tabs story](https://react.carbondesignsystem.com/?path=/story/components-tabs--default) and running an a11y check on it using the [axe Chrome add-on](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd).

Here is an explanation of why this causes problems to users of assistive technologies: https://accessibleweb.com/question-answer/why-are-nested-interactive-controls-an-accessibility-issue/.

**Comment by tw15egan** [comment]
@btepe It seems the only reason it is not being thrown is because of `aria-hidden="true"`. Regardless, this is an incorrect pattern. I have a solution and will push up a PR shortly.
